### PR TITLE
[ACM-10812]: fix addon status not reported in hub

### PR DESCRIPTION
**Pull request by thibaultmg**
This change introduces error specific retry logic to enhance the robustness and reliability of the reconciliation process:

- If a transient error is detected, the event is requeued.
- If the error indicates a potential issue with the kubeconfig of the hub, the controller reloads the config and requeues the event.

To be noted: the controller runtime applies some backoff logic when requeue is requested.

There is still a scenario where the addon status on the hub might remain stale if the spoke addon status is updated before the status controller has fully started. Given that both the endpoint operator (which writes the status) and the status controller start in parallel under the same manager, this scenario has low probability.

If this stale issue continues, we'll have to address this point.
Let's see how it goes with current improvements.

I also used this development as an occasion to refactor some related parts.

### CODE REVIEW

**Review by douglascamata — APPROVED** [review]
/lgtm

**Comment by douglascamata** [comment]
BTW my comments are non-blocking and they can be tackled later, that's why I also approved this even with my questions/points.

File: operators/endpointmetrics/pkg/util/client.go [hunk 1]
```diff
@@ -5,51 +5,65 @@
...
-
 	oav1beta1 "github.com/stolostron/multicluster-observability-operator/operators/multiclusterobservability/api/v1beta1"
 )
 
-const (
-	obAddonName = "observability-addon"
-)
+// HubClientWithReload is a wrapper around the hub client that allows reloading the client.
+// This is useful when the kubeconfig file is updated.
+type HubClientWithReload struct {
```
[the remark this answers is not in the document]
Comment (thibaultmg) [reply operators/endpointmetrics/pkg/util/client.go:22]: done

File: operators/endpointmetrics/pkg/util/client_test.go
[the remark this answers is not in the document]
Comment (thibaultmg) [reply operators/endpointmetrics/pkg/util/client_test.go:1]: It was testing a function I removed. The one fetching the addon in the hub. I didn't feel the need to test what is remaining...

File: operators/endpointmetrics/pkg/util/lease.go [hunk 2]
```diff
@@ -20,8 +21,10 @@ const (
 )
 
 var (
-	namespace   = os.Getenv("WATCH_NAMESPACE")
-	clusterName = os.Getenv("HUB_NAMESPACE")
+	namespace         = os.Getenv("WATCH_NAMESPACE")
+	clusterName       = os.Getenv("HUB_NAMESPACE")
+	log               = ctrl.Log.WithName("util")
```
[the remark this answers is not in the document]
Comment (thibaultmg) [reply operators/endpointmetrics/pkg/util/lease.go:26]: They were declared in the `client.go`. Even if i have remove all dependencies in it, these variables are still needed in other files. 
Will need further refactoring. But I have to stop somewhere. I was already deep into the rabbit hole 😄

**Pull request merged**